### PR TITLE
boost: Extract names of ICU libraries from a pkg-config configuration file

### PR DESCRIPTION
**Pull request by slavaandrejev**
Names of libraries should be configured either using pkg-config or CMake configuration modules. In the case of ICU, there are pkg-config configuration files provided by the ICU maintainers. This change uses pkg-config to extract names of ICU libraries from the configuration files.

### CODE REVIEW

**Comment by lazka** [comment]
Seems a bit brittle to me, is that really worth it?

File: mingw-w64-boost/PKGBUILD [hunk 1]
```diff
@@ -108,6 +108,14 @@ setb2args() {
         local variant=release
     fi
 
+    # Extract names of ICU libraries from the pkg-config configuration file.
+    # Note that the acual order of the libraries in the array is irrelevant.
+    # They are all linked in the same way.
+    local -a icu_libs
+    IFS=" " read -r -a icu_libs <<< "$(pkgconf --libs icu-i18n)"
```
Comment (lazka) [inline mingw-w64-boost/PKGBUILD:115]:
```suggestion
    IFS=" " read -r -a icu_libs <<< "$(pkgconf --libs-only-l icu-i18n)"
```

**Comment by slavaandrejev** [comment]
> Seems a bit brittle to me, is that really worth it? In which case would this help?

This is how it has to be done. We shouldn't “guess” and hard-code the names of the libraries. For example, on my system the names are: icui18n, icuuc, icudata. ICU libraries in boost has always been a huge pane in the rear. You can see it from the last change: after decades of existence, boost is still fixing stuff related to ICU. The reason is simple: boost ignores the maintainer provided configuration files. This change will help keeping things right for the future.

**Comment by lazka** [comment]
yeah, but `pkgconf --libs` could also start outputting dependencies with some future release, breaking this code, and the order is not defined, which is what I mean with brittle.

**Comment by slavaandrejev** [comment]
> the order is not defined

The order is irrelevant, that's another strange thing about what boost is doing. There are three libraries you have to link to, period. There is no need to write separate code for the three.

> could also start outputting dependencies with some future release

It's still better than it was. I am open to suggestions how to improve it.

**Comment by slavaandrejev** [comment]
If there will be extra dependencies, it will break the existing boost build system for sure. We won't make things worse than they are. But this change improves them.

**Comment by slavaandrejev** [comment]
@lazka Hi Christoph. What's the fate of this PR going to be?

**Comment by lazka** [comment]
tbh, I still don't see the point in adding code that extracts names that wont change in the future.

But if you insist I'll merge it.

**Comment by slavaandrejev** [comment]
> tbh, I still don't see the point in adding code that extracts names that wont change in the future.

It's a design principle of having less assumptions. We've learned it a hard way that assumptions lead to failures. The classic nowadays example is the maiden flight of Ariane 5 in 1996. A simple assumption that some number will never exceed the maximum value caused 300 million euros.

Yes, if you don't mind, please merge it.

**Comment by lazka** [comment]
You are adding undocumented assumptions and removing documented ones, so I don't get that argument.

Anyway, fine.

**Pull request merged**